ouroboros-consensus/tools/db-analyse/Main.hs

```diff
@@ -141,6 +146,40 @@ relativeSlotNo epochInfo (SlotNo absSlot) = do
     SlotNo first <- epochInfoFirst epochInfo epoch
     return (epoch, absSlot - first)
 
+{-------------------------------------------------------------------------------
+  Analysis: show the block header size in bytes for all blocks
+-------------------------------------------------------------------------------}
+
+showBlockHeaderSize :: Analysis
+showBlockHeaderSize immDB epochInfo rr = do
+    maxBlockHeaderSizeRef <- newIORef 0
+    processAll immDB rr (go maxBlockHeaderSizeRef)
+    maxBlockHeaderSize <- readIORef maxBlockHeaderSizeRef
+    putStrLn ("Maximum encountered block header size = " <> show maxBlockHeaderSize)
+  where
+    go :: IORef Natural -> ByronBlock -> IO ()
+    go maxBlockHeaderSizeRef blk =
+        case blk of
+          Byron.ByronBlock (Chain.ABOBBlock regularBlk) _ _ ->
+            go' maxBlockHeaderSizeRef (blockSlot blk) regularBlk
+          _otherwise ->
+            return () -- Skip EBBs
+
+    go' :: IORef Natural -> SlotNo -> Chain.ABlock ByteString -> IO ()
```

Comment:
If you have both `go` and `go'`, that means you need to give them better names. Alternatively, inline `go'` into `go`.